src/inspector/view/qml/MuseScore/Inspector/notation/barlines/BarlineSettings.qml

```diff
@@ -56,19 +56,27 @@ Column {
         navigationPanel: root.navigationPanel
         navigationRowStart: root.navigationRowStart
 
-        model: [
-            { text: qsTrc("symUserNames", "Single barline"), value: BarlineTypes.TYPE_NORMAL },
-            { text: qsTrc("symUserNames", "Double barline"), value: BarlineTypes.TYPE_DOUBLE },
-            { text: qsTrc("symUserNames", "Left (start) repeat sign"), value: BarlineTypes.TYPE_START_REPEAT },
-            { text: qsTrc("symUserNames", "Right (end) repeat sign"), value: BarlineTypes.TYPE_END_REPEAT },
-            { text: qsTrc("symUserNames", "Right and left repeat sign"), value: BarlineTypes.TYPE_END_START_REPEAT },
-            { text: qsTrc("symUserNames", "Dashed barline"), value: BarlineTypes.TYPE_DASHED },
-            { text: qsTrc("symUserNames", "Final barline"), value: BarlineTypes.TYPE_FINAL },
-            { text: qsTrc("symUserNames", "Dotted barline"), value: BarlineTypes.TYPE_DOTTED },
-            { text: qsTrc("symUserNames", "Reverse final barline"), value: BarlineTypes.TYPE_REVERSE_END },
-            { text: qsTrc("symUserNames", "Heavy barline"), value: BarlineTypes.TYPE_HEAVY },
-            { text: qsTrc("symUserNames", "Heavy double barline"), value: BarlineTypes.TYPE_DOUBLE_HEAVY },
-        ]
+        visible: root.barlineSettingsModel && root.barlineSettingsModel.isStyleChangingAllowed
+
+        function getModel() {
+            let model = [
+                        { text: qsTrc("symUserNames", "Single barline"), value: BarlineTypes.TYPE_NORMAL },
+                        { text: qsTrc("symUserNames", "Double barline"), value: BarlineTypes.TYPE_DOUBLE },
+                        { text: qsTrc("symUserNames", "Left (start) repeat sign"), value: BarlineTypes.TYPE_START_REPEAT },
+                        { text: qsTrc("symUserNames", "Right (end) repeat sign"), value: BarlineTypes.TYPE_END_REPEAT },
+                        { text: qsTrc("symUserNames", "Right and left repeat sign"), value: BarlineTypes.TYPE_END_START_REPEAT },
+                        { text: qsTrc("symUserNames", "Dashed barline"), value: BarlineTypes.TYPE_DASHED },
+                        { text: qsTrc("symUserNames", "Final barline"), value: BarlineTypes.TYPE_FINAL },
+                        { text: qsTrc("symUserNames", "Dotted barline"), value: BarlineTypes.TYPE_DOTTED },
+                        { text: qsTrc("symUserNames", "Reverse final barline"), value: BarlineTypes.TYPE_REVERSE_END },
+                        { text: qsTrc("symUserNames", "Heavy barline"), value: BarlineTypes.TYPE_HEAVY },
+                        { text: qsTrc("symUserNames", "Heavy double barline"), value: BarlineTypes.TYPE_DOUBLE_HEAVY },
+                    ]
+            return model.filter(function(item) { return root.barlineSettingsModel && root.barlineSettingsModel.isStyleChangingAllowed 
+                && root.barlineSettingsModel.shouldShowType(item.value) })
+        }
```

Comment:
I had hoped this could be simpler, but in order to ensure it does get called every time as needed and worked in all cases, this was the simplest I could make it...if I understood better how Qt Quick translated between javascript and C++ I'm sure it could be improved.

Reply:
You could create a Q_PROPERTY in C++ with type QVariantList, and do the filtering in C++. The getter might look somewhat like this:
```c++
QVariantList result;

for (BarLineType type: theBarlineTypes) {
    if (shouldInclude(type)) {
        QVariantMap map;
        map["value"] = static_cast<int>(type);
        map["text"] = getTheUserNameAsQStringForType(type);
        result << map;
    }
}

return result;
```
This will be translated in QML to a JS array containing JS objects just like when you define it in QML.